src/negate.c

```diff
@@ -0,0 +1,20 @@
+#include "data.table.h"
+
+void negateByRef(SEXP x) {
+  if(TYPEOF(x) != LGLSXP) {
+    error("not logical or integer vector");
+  }
+  int n = length(x);
+  int *ansd = INTEGER(x);
+  for(int i=0;i<n;i++) {
+    if (ansd[i] != NA_LOGICAL) ansd[i] ^= 1;
+  }
+}
+
+
+SEXP notchin(SEXP x, SEXP table) {
+  SEXP result = chin(x, table);
```

Comment:
Needs a PROTECT here. Because something inside negateByRef could trigger a GC which could free the result. It's hard to see why calling INTEGER() could possibly cause a GC (since it's just a pointer offset) but it is an R API call, and we can't be sure about any R API call. Sometimes every entry point in the R API is forced to GC (R can be compiled to do that to debug) and that might be why INTEGER() could GC. Or perhaps the object is an ALTREP and needs to be materialized which could GC. Regardless, `Rchk`'s static analysis would detect the unprotected return so this line wouldn't pass release procedures at that stage.